capsules/src/net/udp/driver.rs

```diff
@@ -466,154 +436,176 @@ impl<'a> LegacyDriver for UDPDriver<'a> {
     ///        the current implementation of this only allows for each app to bind to a single
     ///        port at a time, as such an implementation conserves memory (and is similar
-    ///        to the approach applied by TinyOS and Riot). Further, there is
-    ///        currently no mechanism for anything in the kernel to bind to ports, and there
-    ///        is no distinction between ephemeral ports and reserved ports.
-    /// - `4`: Returns the maximum payload that can be transmitted by apps using this driver.
+    ///        to the approach applied by TinyOS and Riot).
+    ///        /// - `4`: Returns the maximum payload that can be transmitted by apps using this driver.
     ///        This represents the size of the payload buffer in the kernel. Apps can use this
     ///        syscall to ensure they do not attempt to send too-large messages.
 
-    fn command(&self, command_num: usize, arg1: usize, _: usize, appid: AppId) -> ReturnCode {
+    fn command(&self, command_num: usize, arg1: usize, _: usize, appid: AppId) -> CommandResult {
         match command_num {
-            0 => ReturnCode::SUCCESS,
+            0 => CommandResult::success(),
 
             //  Writes the requested number of network interface addresses
             // `arg1`: number of interfaces requested that will fit into the buffer
-            1 => self.do_with_cfg_mut(appid, arg1 * mem::size_of::<IPAddr>(), |cfg| {
-                let n_ifaces_to_copy = cmp::min(arg1, self.interface_list.len());
-                let iface_size = mem::size_of::<IPAddr>();
-                for i in 0..n_ifaces_to_copy {
-                    cfg[i * iface_size..(i + 1) * iface_size]
-                        .copy_from_slice(&self.interface_list[i].0);
-                }
-                // Returns total number of interfaces
-                ReturnCode::SuccessWithValue {
-                    value: self.interface_list.len(),
+            1 => {
+                let res = self
+                    .apps
+                    .enter(appid, |app, _| {
+                        app.app_cfg.mut_map_or(Err(ErrorCode::INVAL), |cfg| {
+                            if cfg.len() != arg1 * mem::size_of::<IPAddr>() {
+                                return Err(ErrorCode::INVAL);
+                            }
+                            let n_ifaces_to_copy = cmp::min(arg1, self.interface_list.len());
+                            let iface_size = mem::size_of::<IPAddr>();
+                            for i in 0..n_ifaces_to_copy {
+                                cfg[i * iface_size..(i + 1) * iface_size]
+                                    .copy_from_slice(&self.interface_list[i].0);
+                            }
+                            // Returns total number of interfaces
+                            Ok(self.interface_list.len())
+                        })
+                    })
+                    .unwrap_or_else(|err| Err(err.into()));
```

Comment:
Why do you have the `match` below? I think you could just return the `CommandResult` from the `.enter(` closure and `.unwrap_or_else` fallback?

Reply:
I was just following the porting guide example, but you're right that's much better. I'll make that change throughout here and submit a PR to update the porting guide to do the same.

Reply:
I'll happily create the porting guide PR if you like.

Reply:
That would be great!

Reply:
This is fixed in this PR now.